tests/beaker_tests/cisco_bgp_neighbor_af/test_bgpneighboraf.rb

```diff
@@ -143,550 +70,242 @@
-  test_harness_bgp_nbr_af(tests, id)
-
-  # -----------------------------------
-  tests[id][:desc] = '3.2 Title Patterns'
-  tests[id][:title_pattern] = '2 blue'
-  tests[id][:af] = { :neighbor => '2.2.2.2', :afi => 'ipv4',
-                     :safi => 'unicast' }
-  test_harness_bgp_nbr_af(tests, id)
-
-  # -----------------------------------
-  tests[id][:desc] = '3.3 Title Patterns'
-  tests[id][:title_pattern] = '2 green 3.3.3.3'
-  tests[id][:af] = { :afi => 'ipv4', :safi => 'unicast' }
-  test_harness_bgp_nbr_af(tests, id)
-
-  # -----------------------------------
-  tests[id][:desc] = '3.4 Title Patterns'
-  tests[id][:title_pattern] = '2 red 4.4.4.4 ipv4'
-  tests[id][:af] = { :safi => 'unicast' }
-  test_harness_bgp_nbr_af(tests, id)
-
-  # -----------------------------------
-  tests[id][:desc] = '3.5 Title Patterns'
-  tests[id][:title_pattern] = '2 yellow 5.5.5.5 ipv4 unicast'
-  tests[id].delete(:af)
-  test_harness_bgp_nbr_af(tests, id)
-
-  # -------------------------------------------------------------------
-  logger.info("\n#{'-' * 60}\nSection 4. L2VPN Default Property Testing")
-  node_feature_cleanup(agent, 'bgp')
-
-  # -----------------------------------
-  id = 'default_properties_l2vpn'
-  test_harness_bgp_nbr_af(tests, id)
+  resource_absent_cleanup(agent, 'cisco_bgp', 'BGP CLEAN :: ')
+  title = '2 blue 1.1.1.1 ipv4 unicast'
+  [
+    :non_def_A1,
+    :non_def_A2,
+    :non_def_A3,
+    :non_def_D1,
+    :non_def_D2,
+    :non_def_M,
+    :non_def_N,
+    :non_def_S1,
+    :non_def_S2,
+    :non_def_S3,
+    :non_def_S4,
+    :non_def_W,
+    :non_def_vrf_only,
+    :non_def_misc_maps_1,
+    :non_def_misc_maps_2,
+    :non_def_misc_maps_3,
+  ].each do |id|
+    tests[id][:title_pattern] = title
+    test_harness_run(tests, id)
+  end
 
-  tests[id][:ensure] = :absent
-  test_harness_bgp_nbr_af(tests, id)
+  test_harness_run(tests, :non_def_ebgp_only)
+  test_harness_run(tests, :non_def_ibgp_only)
 
   # -------------------------------------------------------------------
-  logger.info("\n#{'-' * 60}\nSection 5. L2VPN  Non Default Property Testing")
-  node_feature_cleanup(agent, 'bgp')
-  test_harness_bgp_nbr_af(tests, 'non_default_properties_A1_l2vpn')
-  test_harness_bgp_nbr_af(tests, 'non_default_properties_M_l2vpn')
-  test_harness_bgp_nbr_af(tests, 'non_default_properties_S1_l2vpn')
-  test_harness_bgp_nbr_af(tests, 'non_default_properties_S2_l2vpn')
-  test_harness_bgp_nbr_af(tests, 'non_default_properties_S3_l2vpn')
-
-  # Special Cases
-  test_harness_bgp_nbr_af(tests, 'non_default_properties_ibgp_only_l2vpn')
-  test_harness_bgp_nbr_af(tests, 'non_default_misc_maps_part_1_l2vpn')
+  logger.info("\n#{'-' * 60}\nSection 3. L2VPN Property Testing")
+  resource_absent_cleanup(agent, 'cisco_bgp', 'BGP CLEAN :: ')
+  title = '2 default 1.1.1.1 l2vpn evpn'
+  [
+    # Commented test cases are not supported by l2vpn/evpn
+    # :non_def_A2,
+    # :non_def_A3,
```

Comment:
I meant to have some indication that these are explicitly skipped. But I'm fine with removing them.